python/ql/src/experimental/semmle/python/frameworks/Stdlib.qll

```diff
@@ -9,3 +9,95 @@ private import semmle.python.dataflow.new.TaintTracking
 private import semmle.python.dataflow.new.RemoteFlowSources
 private import experimental.semmle.python.Concepts
 private import semmle.python.ApiGraphs
+
+private module NoSQL {
+  private class PyMongoMethods extends string {
+    // These are all find-keyword relevant PyMongo collection level operation methods
+    PyMongoMethods() {
+      this in [
+          "find", "find_raw_batches", "find_one", "find_one_and_delete", "find_and_modify",
+          "find_one_and_replace", "find_one_and_update"
+        ]
+    }
+  }
+
+  private class PyMongoClientCall extends DataFlow::CallCfgNode, NoSQLQuery::Range {
+    PyMongoClientCall() {
+      this =
+        API::moduleImport("pymongo")
+            .getMember("MongoClient")
+            .getReturn()
+            .getAMember*()
+            .getMember(any(PyMongoMethods pyMongoMethod))
+            .getACall()
+    }
+
+    override DataFlow::Node getQuery() { result = this.getArg(0) }
+  }
+
+  private class PyMongoFlaskMethods extends string {
+    PyMongoFlaskMethods() { this in ["find_one_or_404", any(PyMongoMethods pyMongoMethod)] }
+  }
+
+  private class PyMongoFlaskCall extends DataFlow::CallCfgNode, NoSQLQuery::Range {
+    PyMongoFlaskCall() {
+      this =
+        API::moduleImport("flask_pymongo")
+            .getMember("PyMongo")
+            .getReturn()
+            .getAMember*()
+            .getMember(any(PyMongoFlaskMethods pyMongoFlaskMethod))
+            .getACall()
+    }
+
+    override DataFlow::Node getQuery() { result = this.getArg(0) }
+  }
+
+  private class MongoEngineObjectsCall extends DataFlow::CallCfgNode, NoSQLQuery::Range {
+    MongoEngineObjectsCall() {
+      this =
+        API::moduleImport("mongoengine")
+            .getMember("Document")
+            .getASubclass()
+            .getMember("objects")
+            .getACall()
+    }
+
+    override DataFlow::Node getQuery() { result = this.getArgByName(_) }
+  }
+
+  private class MongoEngineObjectsFlaskCall extends DataFlow::CallCfgNode, NoSQLQuery::Range {
+    MongoEngineObjectsFlaskCall() {
+      this =
+        API::moduleImport("flask_mongoengine")
+            .getMember("MongoEngine")
+            .getReturn()
+            .getMember("Document")
+            .getASubclass()
+            .getMember("objects")
+            .getACall()
+    }
+
+    override DataFlow::Node getQuery() { result = this.getArgByName(_) }
+  }
+
+  private class MongoSanitizerCall extends DataFlow::CallCfgNode, NoSQLSanitizer::Range {
+    MongoSanitizerCall() {
+      this =
+        API::moduleImport("mongosanitizer").getMember("sanitizer").getMember("sanitize").getACall()
+    }
+
+    override DataFlow::Node getSanitizerNode() { result = this.getArg(0) }
+  }
+
+  private class BsonObjectIdCall extends DataFlow::CallCfgNode, NoSQLSanitizer::Range {
```

Comment:
I like the explanation in the commit message from https://github.com/github/codeql/pull/5612/commits/62f3e8d64a4f75fc6fdf21dbeafaa66685f91e39, so let's keep that for future readers as well as a QLDoc :wink: 

```suggestion
  /** ObjectId returns a string representing an id.
   * If at any time ObjectId can't parse it's input (like when a tainted dict in passed in),
   * then ObjectId will throw an error preventing the query from running.
   */
  private class BsonObjectIdCall extends DataFlow::CallCfgNode, NoSQLSanitizer::Range {
```

Reply:
Haha, before my stint as a security researcher I was a Rails dev and it was burned into my skull that helpful git commit messages are paramount. I'll add a QLDoc!

Reply:
That's a very good practice :muscle: :heart: (just in case you're not familiar with such suggestions, you should be able to accept my suggestion, which will then be automatically committed to this PR)

Reply:
I know I can :). I think it's important for me to write out the change manually. That way it enforces me to learn the better practice.